### DISCUSSION TITLE
If I modify some libsonnet file, what should I do when I upgrade.

### DISCUSSION THREAD

**Discussion by hwdef** [prometheus-operator/kube-prometheus#835] [General]
for example, if I modify `kube-prometheus/jsonnet/kube-prometheus/prometheus-adapter/prometheus-adapter.libsonnet`  according to my needs.Later, I want to follow the upstream to upgrade to the new version of kube-prometheus, what should I do?

**Comment by paulfantom** [comment]
You shouldn't modify this file since kube-prometheus is meant to be used as a library. More in https://github.com/prometheus-operator/kube-prometheus#customizing-kube-prometheus
